ports/stm32/mpconfigboard_common.h

```diff
@@ -116,7 +116,7 @@
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1fff7a10)
 #define PYB_EXTI_NUM_VECTORS (23)
 #define MICROPY_HW_MAX_TIMER (14)
-#define MICROPY_HW_MAX_UART (6)
+#define MICROPY_HW_MAX_UART (8)
```

Comment:
MAX_UART adds root pointer entries so this would need to be configured depending if the MCU.  Maybe it would be cleaner to do it like this:
```
#ifdef UART8
#define MICROPY_HW_MAX_UART (8)
#else
#define MICROPY_HW_MAX_UART (6)
#endif
```
That could be used for all MCU series, not just F4 (and so only needs to be written once in this config file).